frontend/src/composables/useShootDns.js

```diff
@@ -177,15 +192,7 @@ export const useShootDns = (manifest, options) => {
       secretName: resourceName, // resourceName is the secret name
     }
     if (isEmpty(dnsServiceExtensionProviders.value)) {
-      setExtension({
-        type: 'shoot-dns-service',
-        providerConfig: {
-          apiVersion: 'service.dns.extensions.gardener.cloud/v1alpha1',
-          kind: 'DNSConfig',
-          syncProvidersFromShootSpecDNS: false,
-          providers: [provider],
-        },
-      })
+      setDnsServiceExtension([provider])
```

Comment:
```suggestion
      setDnsServiceExtensionProviders([provider])
```

Reply:
why would you rename the function. I agree that it makes it clearer to include `providers` in the name as they are passed. But this function not only sets the providers but also the extension itself. I think that this information gets lost by removing it?

Reply:
The only information you pass to this function is the list of providers. The rest is hardcoded and can not be changed. So it does not update the extension but only the provider. As as side effect it populates the hardcoded rest of the extension properties. Both solutions are not optimal. But I wanted to keep the propsal simple.